datasets/dataset_titanic.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+"""
+Titanic dataset.
+(inspired from: https://web.stanford.edu/class/archive/cs/cs109/cs109.1166/problem12.html)
+"""
+
+from sklearn.model_selection import train_test_split
+from keras.models import Sequential
+from keras.layers import Dense, Flatten, Dropout
+from keras.layers import Conv2D, MaxPooling2D
+import requests
+import pandas as pd
+import keras
+import os
+
+# Init dataset-specific variables
+num_classes = 2
+input_shape = (6,)
+
+# Init dataset-specific functions
+def preprocess_dataset_labels(y):
+
+    """
+    Change label to categorical values
+    """
+
+    y = keras.utils.to_categorical(y,num_classes=num_classes)
+    return y
+
+def preprocess_dataset_inputs(x):
+
+    """
+    Return usable boolean values for 'Sex' atribute
+    """
+
+    x["Sex"] = [i=="Male" for i in x["Sex"]]
+    return x.to_numpy()
+
+# download train and test sets
+def load_data():
+
+    """
+    Return a usable dataset
+    """
+
+    url = 'https://web.stanford.edu/class/archive/cs/cs109/cs109.1166/stuff/titanic.csv'
+    r = requests.get(url, allow_redirects=True)
+    open('train.csv', 'wb').write(r.content)
+
+    raw_dataset = pd.read_csv('./train.csv', index_col=False )
+    x = raw_dataset.drop(['Name','Survived'], axis=1)
+    x = preprocess_dataset_inputs(x)
+    y = raw_dataset['Survived']
+    y = y.to_numpy()
+
+    x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.33, random_state=42)
+
+    os.remove("train.csv") # not the cleanest way to deal with temp file
```

Comment:
2 points here:

1. If this isn't the cleanest way, what would it be and can we implement it? 😃 
1. 2 whitespaces before inline comments says PEP8!

Reply:
Thanks for the reference, i will take care of not committing the same mistake.

I also changed the import of the dataset. It's now directly done through read_csv().